app/controllers/api/status_controller.rb

```diff
@@ -11,21 +11,55 @@ def check
     else
       @projects = []
     end
-    render json: @projects, each_serializer: ProjectStatusSerializer, show_score: params[:score], show_stats: internal_api_key?
+
+    render(
+      json: @projects,
+      each_serializer: ProjectStatusSerializer,
+      show_score: params[:score],
+      show_stats: internal_api_key?,
+      project_names: project_names
+    )
   end
 
   private
 
   def find_projects((platform, projects))
-    projects.each_slice(1000).flat_map do |slice|
-      project_find_names = slice.flat_map { |project| project_names(project, platform) }.map(&:downcase)
-      Project.platform(platform).where('lower(platform)=? AND lower(name) in (?)', platform.downcase, project_find_names).includes(:repository, :versions, :repository_maintenance_stats)
+    found = projects.each_slice(1000).flat_map do |slice|
+      project_find_names = slice
+        .flat_map { |project| project_names[project[:name]] }
+        .map(&:downcase)
+      Project
+        .platform(platform)
+        .where('lower(platform)=? AND lower(name) in (?)', platform.downcase, project_find_names)
+        .includes(:repository, :versions, :repository_maintenance_stats)
     end
+
+    if platform.downcase == "go"
+      missing = params[:projects]
+        .select { |project| found.none? { |p| p.name.downcase == project[:name].downcase } }
+
+      missing.each do |project|
+        resolved_name = PackageManager::Go.resolved_name(project[:name])
```

Comment:
Is there any way to batch up these requests to resolve project names?

Reply:
Not that I can think of - we have to fetch each URL